TypeCobol/Compiler/Parser/CodeElementBuilder/CobolStatementsBuilder.cs

```diff
@@ -1435,7 +1435,12 @@ private UnstringStatement.Receiving CreateUnstringReceivingFields(CodeElementsPa
 
         internal CodeElement CreateWhenCondition(CodeElementsParser.WhenConditionContext context) {
 			var statement = new WhenCondition();
-			statement.SelectionObjects = BuildObjectArrayFromParserRules(context.comparisonRHSExpression(), ctx => CreateEvaluateSelectionObject(ctx));
+			var rhsExprs = context.comparisonRHSExpression();
+			statement.SelectionObjects = BuildObjectArrayFromParserRules(rhsExprs, ctx => CreateEvaluateSelectionObject(ctx));
+			if (rhsExprs.Length == 0)
+			{
+				DiagnosticUtils.AddError(statement, "Missing condition in \"when\" clause", context);
```

Comment:
Maybe move that to a CodeElement checker ?

Reply:
I think it can stay here, as these are syntactical errors.

Reply:
We create all of our Diagnostics in checkers (CodeElementCheckers in this case).
It's true we have 3 diagnostics created in CobolStatementsBuilder but this has been done a long time ago and should be moved.

Reply:
Doing this gives unexpected behavior.